forc-pkg/src/pkg.rs

```diff
@@ -2369,14 +2402,66 @@ pub fn build_with_options(build_options: BuildOpts) -> Result<Built> {
     };
 
     let manifest_file = ManifestFile::from_dir(&this_dir)?;
+    let member_manifests = manifest_file.member_manifests()?;
+    let lock_path = manifest_file.lock_path()?;
+    let build_plan = BuildPlan::from_lock_and_manifests(
+        &lock_path,
+        &member_manifests,
+        build_options.pkg.locked,
+        build_options.pkg.offline,
+    )?;
+    let graph = build_plan.graph();
+    let manifest_map = build_plan.manifest_map();
+    let build_profiles: HashMap<String, BuildProfile> = build_plan.build_profiles().collect();
+    // Get the selected build profile using build options
+    let (profile_name, build_profile) = build_profile_from_opts(&build_profiles, &build_options)?;
+
+    // If this is a workspace we want to have all members in the output.
+    let outputs = match &manifest_file {
+        ManifestFile::Package(pkg_manifest) => std::iter::once(
+            build_plan
+                .find_member_index(&pkg_manifest.project.name)
+                .ok_or_else(|| anyhow!("Cannot found project node in the graph"))?,
+        )
+        .collect(),
+        ManifestFile::Workspace(_) => build_plan.member_nodes().collect(),
+    };
+    // Build it!
+    let mut built_workspaces = HashMap::new();
+    let built_packages_with_source_map = build(&build_plan, &build_profile, &outputs)?;
+    let output_dir = build_options
+        .clone()
+        .pkg
+        .output_directory
+        .map(PathBuf::from);
+    for (node_ix, (built_package, source_map)) in built_packages_with_source_map
+        .iter()
+        .filter(|(node_ix, _)| outputs.contains(node_ix))
```

Comment:
We shouldn't have to filter by `outputs` again - we already do that during `build`.

Reply:
I didn't had this kind of check at first but I needed to add this because without this filtering std and core also gets output artifacts even in the scenario where they are implicit depdendencies.

Considering some members of the workspace can depend on non workspace member pkgs (like std and core) without this those non-member dependencies' out folders also gets updated and their bytecode size etc gets printed.

I am on my phone rn so my explanation could be a little off, I will come-up with a cleaner explanation with examples about my concern here once i am back online

Reply:
No worries! I think the reason why dependencies like `std` and `core` were showing up in the output of `build` is because we were missing this:

https://github.com/FuelLabs/sway/pull/3146#discussion_r1014923420

Reply:
Ohh, yeah that makes a lot more sense! Thanks